core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -377,12 +377,36 @@ sealed trait ZIO[-R, +E, +A] extends Serializable { self =>
   final def doUntil(f: A => Boolean): ZIO[R, E, A] =
     repeat(Schedule.doUntil(f))
 
+  /**
+   * Repeats this effect until its result is equal to the predicate.
+   */
+  final def doUntilEquals[A1 >: A](a: => A1): ZIO[R, E, A1] =
```

Comment:
I believe you can omit the type bound here. Also, as Adam pointed out, `doUntilEquals` accepts parameter by value.

Reply:
Ah, scratch it, forgot about the variance. Laziness it is :).